packages/config-loader/src/lib/transform/substitution.ts

```diff
@@ -29,8 +29,13 @@ export function createSubstitutionTransform(env: EnvFunc): TransformFunc {
     }
 
     const parts: (string | undefined)[] = input.split(/(?<!\$)\$\{([^{}]+)\}/);
-    for (let i = 1; i < parts.length; i += 2) {
-      parts[i] = await env(parts[i]!.trim());
+    for (let i = 0; i < parts.length; i++) {
+      if (i % 2 === 0) {
+        parts[i] = parts[i]!.replace(/\$\${(.*?)\}/, '${$1}');
+      }
+      if (i % 2 === 1) {
+        parts[i] = await env(parts[i]!.trim());
+      }
     }
```

Comment:
```suggestion
    const parts: (string | undefined)[] = input.split(/(\$?\$\{[^{}]+\})/);
    for (let i = 1; i < parts.length; i += 2) {
      const part = parts[i]!;
      if (part.startsWith('$$')) {
        parts[i] = part.slice(1);
      } else {
        parts[i] = await env(part.slice(2, -1).trim());
      }
    }
```

Thinking it's nice to skip the extra replace

Reply:
Makes sense.  I applied this, but had to change the `+` in the regex to `*`, otherwise the test for `$${}` being escaped to `${}` failed, as it has no characters in the brackets.

I added 2 additional tests for this as well:
* Test to make sure that `${}` returns `undefined`, since this will now match with the updated regex.
* Test to make sure that multiple escapes in a row are handled correctly.